pkg/kubelet/cm/container_manager.go

```diff
@@ -25,15 +25,19 @@ type ContainerManager interface {
 	// Runs the container manager's housekeeping.
 	// - Ensures that the Docker daemon is in a container.
 	// - Creates the system container where all non-containerized processes run.
-	Start(NodeConfig) error
+	Start() error
 
 	// Returns resources allocated to system containers in the machine.
 	// These containers include the system and Kubernetes services.
 	SystemContainersLimit() api.ResourceList
+
+	// Returns a NodeConfig that is being used by the container manager.
+	GetNodeConfig() NodeConfig
 }
 
 type NodeConfig struct {
-	DockerDaemonContainerName string
-	SystemContainerName       string
-	KubeletContainerName      string
+	RuntimeContainerName string
+	SystemContainerName  string
+	KubeletContainerName string
```

Comment:
I find KubeletContainerName much easier to understand than the old label.
Can we mark the resource container name flag deprecated and just have
Kubelet container name?  Until flag is removed we could keep both values in
synch?

On Tuesday, February 9, 2016, Vish Kannan <notifications@github.com> wrote:

> In pkg/kubelet/cm/container_manager.go
> <https://github.com/kubernetes/kubernetes/pull/20687#discussion_r52393199>
> :
>
> >  }
> >
> >  type NodeConfig struct {
> > -	DockerDaemonContainerName string
> > -	SystemContainerName       string
> > -	KubeletContainerName      string
> > +	RuntimeContainerName string
> > +	SystemContainerName  string
> > +	KubeletContainerName string
>
> Ideally, the flag name has to change. Don't we have a flag deprecation
> policy? Can I just rename a flag? With this change, the internal code is
> more readable though.
> … <#-1321304034_>
> On Tue, Feb 9, 2016 at 3:30 PM, Dawn Chen <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote: In
> pkg/kubelet/cm/container_manager.go <#20687 (comment)
> <https://github.com/kubernetes/kubernetes/pull/20687#discussion_r52392670>>
> : > } > > type NodeConfig struct { > - DockerDaemonContainerName string > -
> SystemContainerName string > - KubeletContainerName string > +
> RuntimeContainerName string > + SystemContainerName string > +
> KubeletContainerName string s/KubeletContainerName/ResourceContainerName to
> keep the name consistent with the flag? Or change the flag name to
> kubelet-container? — Reply to this email directly or view it on GitHub <
> https://github.com/kubernetes/kubernetes/pull/20687/files#r52392670>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20687/files#r52393199>.
>